### PR TITLE
Add JSON output to some of the tools

### PR DESCRIPTION
**Pull request by alexsporn**
Implements #1095

### CODE REVIEW

File: pkg/toolset/toolset.go [hunk 1]
```diff
@@ -26,24 +26,24 @@ const (
...
+func ShouldHandleTools() bool {
 	args := os.Args[1:]
 
-	toolFound := false
 	for i, arg := range args {
 		if strings.ToLower(arg) == "tool" || strings.ToLower(arg) == "tools" {
 			args = args[i:]
-			toolFound = true
+			return true
 			break
```
Comment (muXxer) [inline pkg/toolset/toolset.go]: no need for the break anymore

File: core/app/app.go [hunk 2]
```diff
@@ -97,8 +100,11 @@ Command line flags:
 		flag.PrintDefaults()
 	}
 
-	parseFlags(flagSets)
-	printVersion(flagSets)
+	if !toolsRequested {
+		// No tools requested, so parse all flags as config flags
+		parseFlags(flagSets)
```
Comment (muXxer) [inline core/app/app.go]: Wouldn't that imply that we can't pass the correct config to the tools anymore?
For example I want to run the tools with a config_devnet.json, I can't call `-c config_devnet.json` anymore?

The config is for example needed for the tools ToolP2PIdentityGen, ToolP2PExtractIdentity, ToolJWTApi.

File: pkg/toolset/ed25519.go [hunk 3]
```diff
@@ -2,68 +2,115 @@ package toolset
...
 )
 
+type keys struct {
+	PublicKey      string `json:"publicKey"`
+	PrivateKey     string `json:"privateKey,omitempty"`
+	Ed25519Address string `json:"ed25519"`
+	Bech32Address  string `json:"bech32"`
+}
+
+func print(pubKey ed25519.PublicKey, privKey ed25519.PrivateKey, hrp iotago.NetworkPrefix, outputJSON bool) {
```
Comment (muXxer) [inline pkg/toolset/ed25519.go]: Maybe rename to `printEd25519Info` ? :)

File: pkg/toolset/jwt.go [hunk 4]
```diff
@@ -17,18 +19,23 @@ import (
...
-	printUsage := func() {
-		println("Usage:")
-		println(fmt.Sprintf("	%s [P2P_DATABASE_PATH]", ToolJWTApi))
-		println()
-		println("	[P2P_DATABASE_PATH] - the path to the p2p database folder (optional)")
-		println()
-		println(fmt.Sprintf("example: %s %s", ToolJWTApi, "p2pstore"))
+	fs := flag.NewFlagSet("", flag.ExitOnError)
+	outputJSON := fs.Bool("json", false, "format output as JSON")
+	databasePath := fs.String("database", "", "the path to the p2p database folder (optional)")
```
Comment (muXxer) [inline pkg/toolset/jwt.go:24]: maybe `database.path` ?
Or is this conflicting to our default setting?

Comment (alexsporn) [reply pkg/toolset/jwt.go:24]: why use long flags for the tools? They are long for the config, because we have a complex JSON structure

Comment (muXxer) [reply pkg/toolset/jwt.go:24]: ok

File: core/app/app.go (hunk 2 above)
Comment (alexsporn) [reply core/app/app.go]: Ok, check again

**Review by muXxer — APPROVED** [review]

**Pull request merged**